processing/src/main/java/io/druid/segment/filter/SelectorFilter.java

```diff
@@ -31,25 +39,93 @@
 {
   private final String dimension;
   private final String value;
+  private final ExtractionFn extractionFn;
 
   public SelectorFilter(
       String dimension,
-      String value
+      String value,
+      ExtractionFn extractionFn
   )
   {
     this.dimension = dimension;
     this.value = value;
+    this.extractionFn = extractionFn;
   }
 
   @Override
   public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   {
-    return selector.getBitmapIndex(dimension, value);
+    if (extractionFn == null) {
+      return selector.getBitmapIndex(dimension, value);
+    } else {
+      final List<Filter> filters = makeFilters(selector);
+      if (filters.isEmpty()) {
+        return selector.getBitmapFactory().makeEmptyImmutableBitmap();
+      }
+      return new OrFilter(makeFilters(selector)).getBitmapIndex(selector);
+    }
   }
 
   @Override
   public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
-    return factory.makeValueMatcher(dimension, value);
+    if (extractionFn == null) {
+      return factory.makeValueMatcher(dimension, value);
+    } else {
+      return factory.makeValueMatcher(
+          dimension, new Predicate<String>()
+          {
+            @Override
+            public boolean apply(String input)
+            {
+              // Assuming that a null/absent/empty dimension are equivalent from the druid perspective
+              return value.equals(Strings.nullToEmpty(extractionFn.apply(Strings.emptyToNull(input))));
+            }
+          }
+      );
+    }
+  }
+
+  private List<Filter> makeFilters(BitmapIndexSelector selector)
+  {
+    Indexed<String> allDimVals = selector.getDimensionValues(dimension);
```

Comment:
@navis The code was copied from ExtractionFilter, but agreed, it looks cleaner with Iterable, changed it to that